tokio/src/net/tcp/listener.rs

```diff
@@ -72,15 +72,17 @@ cfg_tcp! {
 }
 
 impl TcpListener {
-    /// Creates a new TcpListener which will be bound to the specified address.
+    /// Creates a new TcpListener, which will be bound to the specified address.
     ///
     /// The returned listener is ready for accepting connections.
     ///
     /// Binding with a port number of 0 will request that the OS assigns a port
     /// to this listener. The port allocated can be queried via the `local_addr`
     /// method.
     ///
-    /// The address type can be any implementor of `ToSocketAddrs` trait.
+    /// The address type can be any implementor of the [`ToSocketAddrs`] trait.
+    /// Note that strings only implement this trait when the **`dns`** feature
+    /// is enabled.
```

Comment:
maybe worth explaining that this will perform DNS lookups for strings containing DNS names? Take it or leave it.